Atividades/EmCasa/exercicios.js

```diff
@@ -0,0 +1,161 @@
+        }
+        else  {
+            return console.log (`Você está acaraciando o animal`)
+        }
+    }
+
+    alimentar(alimentos){
+        if(this instanceof Gato && (alimentos == "sache") || (alimentos == "frango") || (alimentos == "peixe")) { // colocar as comidas como estática nos gatos pq são comuns para todos os gatos?
+            this.social == true
+        }
+        else if(this instanceof Papagaio){
+            return console.log(`este ${this.nome} está ${this.falar()}`)
+        } else {
+            return console.log(`Você aliementou ${this.nome} com ${alimentos}`)
+        }
+    }
+
+   
+}
+
+class Domesticos extends Animal {
+    constructor(nome, idade, cor, castrado){
+        super(nome, idade, cor)
+        this.castrado = castrado
+        this.vacinas = []
+    }
+
+    vacinar(vacina) {
+        this.consultar()
+        this.vacinas.push(vacina) 
+        return this.vacinas
+    }
+
+    castrar() {
+        if(this.castrado == true){
+            return console.log(`${this.nome} já foi castrado`)
+        } else {
+            return this.castrado == true
+        }
+    }
+}
+
+class Gato extends Domesticos {
+    constructor(nome, idade, cor, consultas, castrado, vacinas, externo, social){
+        super(nome, idade, cor, consultas, castrado, vacinas)
+        this.externo = externo
+        this.social = social
+    }
+
+    miar(){
+        return console.log("miau") // não sei porque a função está retornando um "undefined" junto com o "miau"
+    }
+}
+
+class Cachorro extends Domesticos{
+    #ferido
+    
+    constructor(nome, idade, cor, consultas, castrado, vacinas, raca, ferido){
+        super(nome, idade, cor, consultas, castrado, vacinas)
+        this.raca= raca
+        this.#ferido = ferido
+    }
+
+    latir(){
+        console.log("au au")
+    }
+
+     brincar(){
+         if(this.ferido == true){
+            console.log(`${this.nome} está ferido e não consegue brincar`)
+         } else {
+            console.log(`Você brincou com ${this.nome}`)
+         }
+     }
+}
+
+class Hamster extends Animal{
+    constructor(nome, idade, cor, consultas, tipo){
+    super(nome, idade, cor, consultas)
+        this.tipo = tipo
+    }
+
+    consultar(){
+        this.brincar()
+        console.log("Agora que você brincou com o Hamster, pode seguir com a consulta")
+    }
+        
+}
+
+class Papagaio extends Animal{
+    constructor(nome, idade, cor, consultas){
+    super(nome, idade, cor, consultas)
+    }
+    falar(){
+        console.log("falando")
+    }
+}
+
+const nina = new Gato("Nina", 1.5, "frajola",[], false, ["raiva"], false, false)
+const zeus = new Cachorro()
+const hamtaro = new Hamster()
+const loro = new Papagaio("Loro", 20, "verde", [])
+
+
+console.log(nina.miar()) // retorna miau e um "undefined" que não sei da onde vem
+console.log(nina.vacinar("teste"))
+console.log(nina.vacinas) // está retornando só a vacina teste, não retorna a vacina raiva passada no nina
```

Comment:
Quando vc cria a nina e passa a vacina "raiva", esse valor não é guardado em lugar algum. 
Lembra que, por nina ser uma instância de Gato, "vacinas" está sendo passado pro super de Gato, mas como não existe esse parâmetro no construtor do Domestico (classe que Gato estende), ele não chega pra dentro da função:

class Gato extends Domesticos {
    constructor(nome, idade, cor, consultas, castrado, vacinas, externo, social){ 
        super(nome, idade, cor, consultas, castrado, vacinas) //passou vacinas aqui
        this.externo = externo
        this.social = social
    }

```
class Domesticos extends Animal {
    constructor(nome, idade, cor, castrado) // mas vacinas não chega aqui
    {
        super(nome, idade, cor)
        this.castrado = castrado
        this.vacinas = [] 
    }
    ...
```
    
 Mas, ainda que chegasse, na linha 68, this.vacinas recebe uma lista vazia.
 Por isso, na linha 149 nina recebe sua primeira e única vacina: "teste". 
 (O lado bom é que nina está protegida contra "teste" até o ano que vem :P )